spec/Resolver/CompositeAgreementHistoryResolverSpec.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace spec\BitBag\SyliusAgreementPlugin\Resolver;
+
+use BitBag\SyliusAgreementPlugin\Entity\Agreement\AgreementHistoryInterface;
+use BitBag\SyliusAgreementPlugin\Entity\Agreement\AgreementInterface;
+use BitBag\SyliusAgreementPlugin\Repository\AgreementHistoryRepositoryInterface;
+use BitBag\SyliusAgreementPlugin\Resolver\CompositeAgreementHistoryResolver;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\Order\Context\CartContextInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\Security\Core\Security;
+
+final class CompositeAgreementHistoryResolverSpec extends ObjectBehavior
```

Comment:
One scenario is missing :)

Reply:
Done

Reply:
Function is changed, no more scenario is needed